lib/foreman_deployments/engine.rb

```diff
@@ -18,29 +18,39 @@ class Engine < ::Rails::Engine
         # Add permissions
         security_block :foreman_deployments do |map|
           map.permission :view_deployments,
-                         :"foreman_deployments/api/v2/deployments" => [:index, :show]
+                         { :'foreman_deployments/api/v2/deployments' => [:index, :show] },
+                         :resource_type => ForemanDeployments::Deployment.name
           map.permission :create_deployments,
-                         :"foreman_deployments/api/v2/deployments" => [:create]
-          map.permission :edit_deployments,
-                         :"foreman_deployments/api/v2/deployments" => [:update]
+                         { :'foreman_deployments/api/v2/deployments' => [:create, :merge_configuration, :replace_configuration] },
+                         :resource_type => ForemanDeployments::Deployment.name
+          map.permission :run_deployments,
+                         { :'foreman_deployments/api/v2/deployments' => [:run] },
```

Comment:
you'll need to define mapping for custom names in controllers, see https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/hostgroups_controller.rb#L84 for inspiration